File: Habitica/src/main/java/com/habitrpg/android/habitica/models/user/Inbox.java
package com.habitrpg.android.habitica.models.user;

import com.habitrpg.android.habitica.models.social.ChatMessage;

import java.util.ArrayList;
import java.util.List;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.Ignore;
import io.realm.annotations.PrimaryKey;

public class Inbox extends RealmObject {

    @PrimaryKey
    private String userId;

    User user;
    private boolean optOut;
    @Ignore
    private List<Object> blocks = new ArrayList<>();
    private int newMessages;

    /**
     * @return The optOut
     */
    public boolean getOptOut() {
        return optOut;
    }

    /**
     * @param optOut The optOut
     */
    public void setOptOut(boolean optOut) {
        this.optOut = optOut;
    }

    /**
     * @return The blocks
     */
    public List<Object> getBlocks() {
        return blocks;
    }

    /**
     * @param blocks The blocks
     */
    public void setBlocks(List<Object> blocks) {
        this.blocks = blocks;
    }

    /**
     * @return The newMessages
     */
    public int getNewMessages() {
        return newMessages;
    }

    /**
     * @param newMessages The newMessages
     */
    public void setNewMessages(int newMessages) {
        this.newMessages = newMessages;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }
}

File: shared/src/androidMain/kotlin/com/habitrpg/shared/habitica/models/user/Inbox.kt
package com.habitrpg.shared.habitica.models.user

import io.realm.RealmObject
import io.realm.annotations.Ignore
import io.realm.annotations.PrimaryKey
import java.util.*

actual open class Inbox : RealmObject() {

    @PrimaryKey
    actual var userId: String? = null

    internal actual var user: User? = null
    /**
     * @return The optOut
     */
    /**
     * @param optOut The optOut
     */
    actual var optOut: Boolean = false
    /**
     * @return The blocks
     */
    /**
     * @param blocks The blocks
     */
    @Ignore
    actual var blocks: List<Any> = ArrayList()
    /**
     * @return The newMessages
     */
    /**
     * @param newMessages The newMessages
     */
    actual var newMessages: Int = 0
}